pkg/modules/generators/monitoring/monitoring_generator.go

```diff
@@ -44,91 +52,182 @@ func (g *monitoringGenerator) Generate(spec *apiv1.Intent) error {
 				return err
 			}
 		} else {
-			return fmt.Errorf("MonitorType should either be service or pod %s", g.project.Prometheus.MonitorType)
+			return fmt.Errorf("MonitorType should either be service or pod %s", g.app.Monitoring.MonitorType)
 		}
 	}
 
 	return nil
 }
+
+// parseWorkspaceConfig parses the config items for monitoring generator in workspace configurations.
+func (g *monitoringGenerator) parseWorkspaceConfig() error {
+	wsConfig, ok := g.modulesConfig[monitoring.ModuleName]
+	// If AppConfiguration contains monitoring config but workspace does not,
+	// respond with the error ErrEmptyModuleConfigBlock
+	if g.app.Monitoring != nil && !ok {
+		return workspace.ErrEmptyModuleConfigBlock
+	}
+
+	if operatorMode, ok := wsConfig[monitoring.OperatorModeKey]; ok {
+		g.app.Monitoring.OperatorMode = operatorMode.(bool)
+	}
+
+	if monitorType, ok := wsConfig[monitoring.MonitorTypeKey]; ok {
+		g.app.Monitoring.MonitorType = monitoring.MonitorType(monitorType.(string))
+	} else {
+		g.app.Monitoring.MonitorType = monitoring.DefaultMonitorType
+	}
+
+	if interval, ok := wsConfig[monitoring.IntervalKey]; ok {
+		g.app.Monitoring.Interval = prometheusv1.Duration(interval.(string))
+	} else {
+		g.app.Monitoring.Interval = monitoring.DefaultInterval
+	}
+
+	if timeout, ok := wsConfig[monitoring.TimeoutKey]; ok {
+		g.app.Monitoring.Timeout = prometheusv1.Duration(timeout.(string))
+	} else {
+		g.app.Monitoring.Timeout = monitoring.DefaultTimeout
+	}
+
+	if scheme, ok := wsConfig[monitoring.SchemeKey]; ok {
+		g.app.Monitoring.Scheme = scheme.(string)
+	} else {
+		g.app.Monitoring.Scheme = monitoring.DefaultScheme
+	}
+
+	parsedTimeout, err := time.ParseDuration(string(g.app.Monitoring.Timeout))
+	if err != nil {
+		return err
+	}
+	parsedInterval, err := time.ParseDuration(string(g.app.Monitoring.Interval))
+	if err != nil {
+		return err
+	}
+
+	if parsedTimeout > parsedInterval {
+		return monitoring.ErrTimeoutGreaterThanInterval
+	}
+
+	return nil
+}
+
+func (g *monitoringGenerator) buildMonitorObject(monitorType monitoring.MonitorType) (runtime.Object, error) {
+	// If Prometheus runs as an operator, it relies on Custom Resources to
+	// manage the scrape configs. CRs (ServiceMonitors and PodMonitors) rely on
+	// corresponding resources (Services and Pods) to have labels that can be
+	// used as part of the label selector for the CR to determine which
+	// service/pods to scrape from.
+	// Here we choose the label name kusion_monitoring_appname for two reasons:
+	// 1. Unlike the label validation in Kubernetes, the label name accepted by
+	// Prometheus cannot contain non-alphanumeric characters except underscore:
+	// https://github.com/prometheus/common/blob/main/model/labels.go#L94
+	// 2. The name should be unique enough that is only created by Kusion and
+	// used to identify a certain application
+	monitoringLabels := map[string]string{
+		"kusion_monitoring_appname": g.appName,
+	}
+
+	if monitorType == "Service" {
```

Comment:
You could replace this hardcoded value with the const type you have defined before

Reply:
done

Reply:
done